ddtrace/internal/ci_visibility/_api_client.py

```diff
@@ -272,11 +286,57 @@ def _do_request(self, method: str, endpoint: str, payload: str, timeout: t.Optio
             if conn is not None:
                 conn.close()
 
+    def _do_request_with_telemetry(
+        self,
+        method: str,
+        endpoint: str,
+        payload: str,
+        metric_names: APIRequestMetricNames,
+        timeout: t.Optional[float] = None,
+    ) -> t.Any:
+        """Performs a request with telemetry submitted according to given names"""
+        sw = StopWatch()
+        sw.start()
+        error_type: t.Optional[ERROR_TYPES] = None
+        response_bytes: t.Optional[int] = None
+        try:
+            try:
+                response = self._do_request(method, endpoint, payload, timeout=timeout)
+            except _NETWORK_ERRORS:
+                error_type = ERROR_TYPES.TIMEOUT
+                raise
+            if response.status >= 400:
+                error_type = ERROR_TYPES.CODE_4XX if response.status < 500 else ERROR_TYPES.CODE_5XX
+                if response.status == 403:
+                    raise CIVisibilityAuthenticationException()
+                raise ValueError("API response status code: %d", response.status)
+            try:
+                sw.stop()  # Stop the timer before parsing the response
+                response_bytes = len(response.body)
+                if isinstance(response.body, bytes):
+                    parsed = json.loads(response.body.decode())
+                else:
+                    parsed = json.loads(response.body)
```

Comment:
I've seen this repeated elsewhere. It looks like our `Response` class already has a [`get_json()` method](https://github.com/DataDog/dd-trace-py/blob/main/ddtrace/internal/utils/http.py#L272) that kinda does the same thing (although it catches `ValueError` and `TypeError` and returns nothing in that case, and doesn't catch `JSONDecodeError`, which is a weird choice of exceptions to handle).

Reply:
It's vestigial from the original code, and IIRC, one of them was only there for compatibility with Python 2.7 anyway. 

I'll double check and remove one if that's the case. I'm not sure I want to use the `get_json()` method for the reason you mentioned.